src/world.rs

```diff
@@ -110,11 +110,26 @@ impl World {
         self.hosts.insert(addr, Host::new(addr));
     }
 
-    /// Send `message` from `src` to `dst`. Delivery is asynchronous and not
-    /// guaranteed.
+    fn sent_within_current_host(&self, src: SocketAddr, dst: SocketAddr) -> bool {
+        let ip = self.current.expect("current host missing");
+
+        src.ip().is_loopback() && dst.ip().is_loopback()
+            || src.ip().is_loopback() && dst.ip() == ip
+            || src.ip() == ip && dst.ip().is_loopback()
+    }
+
+    /// Send `message` from `src` to `dst`.
+    /// Delivery between hosts is asynchronous and not guaranteed.
     pub(crate) fn send_message(&mut self, src: SocketAddr, dst: SocketAddr, message: Protocol) {
-        self.topology
-            .enqueue_message(&mut self.rng, src, dst, message);
+        // Loopback communication does not use Topology.
+        if self.sent_within_current_host(src, dst) {
+            let _ = self
```

Comment:
This is tricky. Should the communication on loopback be synchronous?
Topology has some queuing mechanism based on ticks, what about here?

Reply:
Good question... I'd think about if two tasks could get into a busy loop chatting back and forth on the same host.

Reply:
In the proposed implementation, the tasks would communicate with each other as long they have something to send within a single step. They would block the current simulation step if they wanted to communicate. Only after they finished, simulation would proceed.

Reply:
I think introducing a queuing mechanism on a host level would solve the issue. Will try to implement that.